data/data/agent/systemd/units/assisted-service.service.template

```diff
@@ -8,10 +8,12 @@ After=network-online.target assisted-service-pod.service
 
 [Service]
 Environment=PODMAN_SYSTEMD_UNIT=%n
+EnvironmentFile=%t/agent-images.env
 Restart=on-failure
 TimeoutStopSec=300
 ExecStartPre=/bin/rm -f %t/%n.ctr-id
-ExecStart=/usr/bin/podman run --cidfile=%t/%n.ctr-id --cgroups=no-conmon --log-driver=journald --rm --pod-id-file=%t/assisted-service-pod.pod-id --sdnotify=conmon --replace -d --name=service -v /opt/agent/tls:/opt/agent/tls:z {{.MirrorRegistriesMount}} {{.CaBundleMount}} --env-file=/usr/local/share/assisted-service/assisted-service.env --env-file=/usr/local/share/assisted-service/images.env --env-file=/etc/assisted-service/node0 quay.io/edge-infrastructure/assisted-service:latest
+# ExecStart=/usr/local/bin/start-assisted-service.sh %t %n agent-installer-api-server
+ExecStart=/usr/bin/podman run --cidfile=%t/%n.ctr-id --cgroups=no-conmon --log-driver=journald --rm --pod-id-file=%t/assisted-service-pod.pod-id --sdnotify=conmon --replace -d --name=service -v /opt/agent/tls:/opt/agent/tls:z {{.MirrorRegistriesMount}} {{.CaBundleMount}} --env-file=/usr/local/share/assisted-service/assisted-service.env --env-file=/usr/local/share/assisted-service/images.env --env-file=/etc/assisted-service/node0 $SERVICE_IMAGE
```

Comment:
Didn't we need to pass some of the image pullspecs to assisted-service in the environment?

Reply:
That is already being done, it is now in get-container-images.sh. The images are set in the env file:
AssistedEnvFile=/usr/local/share/assisted-service/assisted-service.env